.github/workflows/person.rb

```diff
@@ -1,49 +1,36 @@
 require_relative 'Nameable'
-require_relative 'BaseDecorator'
+class Person < Nameable
 
+  attr_accessor :name, :age, :rentals
+  attr_reader :id, 
 
-class Person < Nameable
-    attr_accessor :name, :age
-    attr_reader :id
-  
-    def initialize(name = "Unknown", age = nil, parent_permission = true)
-      @id = generate_id
-      @age = age
-      @name = name
-      @parent_permission = parent_permission
-    end
-  
-    def age=(age)
-      @age = age
-    end
-  
-    def name=(name)
-      @name = name
-    end
-  
-    def can_use_services?
-      of_age? || @parent_permission
-    end
-  
-    def correct_name
-      @name
-    end
-  
-    private
-  
-    def of_age?
-      @age.to_i >= 18
-    end
-  
-    def generate_id
-      rand(1..1000)
-    end
+  def initialize(name = "Unknown", age = nil, parent_permission = true)
+    super(name)
+    @id = generate_id
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def age=(age)
+    @age = age
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```

Comment:
- [ ] Let's make the Person class `add_rental` to be independent of the Book class, so we can use it without having to call it from the book class.

  ```
  def add_rental(date, book)
    Rental.new(date, self, book)
  end
  ```

Reply:
done